rpcs3/rpcs3qt/game_list_frame.cpp

```diff
@@ -506,30 +506,64 @@ void game_list_frame::Refresh(const bool fromDrive, const bool scrollAfter)
 
 				if (fs::is_file(entry_path + "/PS3_DISC.SFB"))
 				{
-					add_disc_dir(entry_path);
+					if (!is_disc)
+					{
+						game_list_log.error("Invalid game path found in %s", entry_path);
+					}
+					else
+					{
+						add_disc_dir(entry_path);
+					}
 				}
 				else
 				{
-					path_list.emplace_back(entry_path);
+					if (is_disc)
+					{
+						game_list_log.error("Invalid disc path found in %s", entry_path);
+					}
+					else
+					{
+						path_list.emplace_back(entry_path);
+					}
 				}
 			}
 		};
 
-		add_dir(_hdd + "game/");
-		add_dir(_hdd + "disc/");
+		add_dir(_hdd + "game/", false);
+		add_dir(_hdd + "disc/", true);
 
 		for (auto pair : YAML::Load(fs::file{fs::get_config_dir() + "/games.yml", fs::read + fs::create}.to_string()))
 		{
 			std::string game_dir = pair.second.Scalar();
+
 			game_dir.resize(game_dir.find_last_not_of('/') + 1);
 
 			if (fs::is_file(game_dir + "/PS3_DISC.SFB"))
 			{
+				// Check if a path loaded from games.yml is already registered in add_dir(_hdd + "disc/");
+				if (game_dir.starts_with(_hdd))
```

Comment:
This should suffice imo
```suggestion
				if (game_dir.starts_with(_hdd + "disc/"))
				{
					game_list_log.trace("Removed duplicate for %s: %s", pair.first.Scalar(), pair.second.Scalar());
					continue;
				}
```

Reply:
nevermind

Reply:
Well, I handled one dumb case when you put games deeply inside another directory in /dev_hdd0/disc/ and expect it to work correctly. Must be user bad, but still.